cMake/FreeCAD_Helpers/PrintFinalReport.cmake

```diff
@@ -1,182 +1,128 @@
+                "not found"
+                "uic: [${PYSIDE_UIC_EXECUTABLE}] rcc: [${PYSIDE_RCC_EXECUTABLE}]"
+    )
     if(FREECAD_USE_FREETYPE)
-        if(FREETYPE_FOUND)
-            report(Freetype ${FREETYPE_VERSION_STRING})
-        else()
-            report(Freetype "not found")
-        endif()
+        conditional(Freetype FREETYPE_FOUND "not found" ${FREETYPE_VERSION_STRING})
     else()
-        report(Freetype "disabled")
+        simple(Freetype "disabled")
     endif()
-    report(OpenGL ${OPENGL_gl_LIBRARY})
-    report(OpenGLU "[${OPENGL_glu_LIBRARY}][${OPENGL_INCLUDE_DIR}]")
-    report(Coin3D "${COIN3D_VERSION} [${COIN3D_LIBRARIES}] [${COIN3D_INCLUDE_DIRS}]")
-    report(Pivy ${PIVY_VERSION})
+    simple(OpenGL_Lib [${OPENGL_gl_LIBRARY}])
+    simple(OpenGLU_Lib [${OPENGL_glu_LIBRARY}])
+    simple(OpenGLU_Incl [${OPENGL_INCLUDE_DIR}])
+    simple(Coin3D "${COIN3D_VERSION} [${COIN3D_LIBRARIES}] [${COIN3D_INCLUDE_DIRS}]")
+    simple(Pivy ${PIVY_VERSION})
     if (WIN32)
-        #report(SPNAV "not available yet for your OS") # FREECAD_USE_3DCONNEXION instead...
-    else()
-        if(SPNAV_FOUND)
-            report(SPNAV "[${SPNAV_LIBRARY}] [${SPNAV_INCLUDE_DIR}]")
-        else()
-            report(SPNAV "not found")
-        endif()
-    endif()
-
-    if(MATPLOTLIB_FOUND)
-        report(Matplotlib ${MATPLOTLIB_VERSION})
+        #simple(SPNAV "not available yet for your OS") # FREECAD_USE_3DCONNEXION instead...
     else()
-        report(Matplotlib "not found")
+        conditional(SPNAV SPNAV_FOUND "not found" "[${SPNAV_LIBRARY}] [${SPNAV_INCLUDE_DIR}]")
     endif()
+    conditional(Matplotlib MATPLOTLIB_FOUND "not found" ${MATPLOTLIB_VERSION})
     if(BUILD_VR)
-        if(RIFT_FOUND)
-            report(Rift ${Rift_VERSION})
-        else()
-            report(Rift "not found")
-        endif()
+        conditional(Rift RIFT_FOUND "not found" ${Rift_VERSION})
     else()
-        report(Rift "not enabled (BUILD_VR)")
+        simple(Rift "not enabled (BUILD_VR)")
     endif()
     if(DOXYGEN_FOUND)
-        report(Doxygen ${DOXYGEN_VERSION})
-        report(Language ${DOXYGEN_LANGUAGE})
-        if(COIN3D_DOC_FOUND)
-            report(Coin3D_DOC "found [${COIN3D_DOC_PATH}]")
-        else()
-            report(Coin3D_DOC "not found")
-        endif()
+        simple(Doxygen ${DOXYGEN_VERSION})
+        simple(Language ${DOXYGEN_LANGUAGE})
+        conditional(Coin3D_DOC COIN3D_DOC_FOUND "not found" ${COIN3D_DOC_PATH})
     else()
-        report(Doxygen "not found")
-    endif()
-    if(BUILD_TEST)
-        report(Tests "ON")
-    else()
-        report(Tests "OFF")
+        simple(Doxygen "not found")
     endif()
+    conditional(Tests BUILD_TEST "OFF" "ON")
+
+    ############## End report data ####################
+
+    message(STATUS "\n==============\n"
+                   "Summary Report\n"
+                   "==============\n")
+
+    list(SORT simpleLines CASE INSENSITIVE)
```

Comment:
I suspect this is going to be unpopular -- at the very least we should have a separate section at the top that lists the most important things first, and together. That is, the four compiler settings, and the version of Python. You might be better off manually sorting above and dropping this line.

Reply:
Respectfully, I ask that we leave sorted for now as is. Its easy to change if real problems arise. 

I have been using this report quite a bit lately during various experiments and unsorted was driving me nuts. In the best traditions of 'data' I scratched the itch and sorted the result. Much better!

In unordered data it makes sense to have the most used at the top, but that is not to say unordered was the right choice in the first place. We are educated to expect sorting. The only important item to the reader is the one being sought, which sorting facilitates (ha ha).

Special cases almost always hint at underlying issues and are the first step to chaos. Every item is important. Being 'special' is subjective and circumstantial. E.g. during initial setup the python version might be of special interest, but after that not.

There are some alternatives:

The names of 'special' cases could simply be prefaced with e.g. '_' to move them up, or could be moved up by other means, but this is a slippery slope. It frustrates the reader by requiring recognition that there are two different systems in place. The first few items is where the brain automatically and subconsiously looks to determine if sorted.

Sort personal choice could be a cache flag. Seems OTT.

Reply:
I certainly agree that the libraries should be sorted alphabetically -- to me it's not so much that any one item is "special" as that this is really two _different_ lists that by virtue of our old formatting _appeared_ to be one. I view this as list of compiler settings, and a list of library versions. 

That said, it is of course possible that simply sorting the whole thing and then being more careful with how we label things (e.g. "Flags" should really be "Compiler Flags"), will still achieve the desired effect. Let me ping a few other Maintainers and see if anyone else has an opinion. If not, I'll merge it as-is.

Reply:
Well, seems like no one else cares (yet 😉). I'm going to merge as-is, with the caveat that I'd still like to find a way to keep the actual compiler settings close to one another.